### PR TITLE
Trees: turn Defn.Var into proper Tree.WithBody

### PR DESCRIPTION
**Pull request by kitbellew**
Do not hide placeholder assignment.

### CODE REVIEW

File: scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala [hunk 1]
```diff
@@ -574,25 +574,36 @@ object Decl {
...
+    def apply(rhs: Option[Term]): Term = rhs.getOrElse(Term.Placeholder())
+    def toOpt(body: Term): Option[Term] = if (isPlaceholder(body)) None else Some(body)
+  }
   @ast class Var(
       mods: List[Mod],
       pats: List[Pat] @nonEmpty,
       decltpe: Option[sm.Type],
-      rhs: Option[Term]
+      @replacesFields("4.7.2", VarRhsCtor)
+      body: Term
```
Comment (tgodzik) [inline scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala:600]: Does `var a: Int` and `var a: Int = _` become the same thing? Should we have something like an `EmptyTree` insteadto models the first case?

Comment (kitbellew) [reply scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala:600]: `var a: Int` becomes De**cl**.Var whereas the other one is De**fn**.Var

Comment (tgodzik) [reply scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala:600]: Ach, right! I forgot about it!

**Review by tgodzik — APPROVED** [review]
LGTM

**Pull request merged**